src/pocketmine/PocketMine.php

```diff
@@ -487,16 +487,31 @@ function cleanPath($path){
 	$killer = new ServerKiller(8);
 	$killer->start();
 
+	$erroredThreads = 0;
+
 	foreach(ThreadManager::getInstance()->getAll() as $id => $thread){
 		$logger->debug("Stopping " . $thread->getThreadName() . " thread");
-		$thread->quit();
+		try{
+			$thread->quit();
+			$logger->debug($thread->getThreadName() . " thread stopped successfully.");
```

Comment:
Is $thread->quit() blocking? I don't think you can assume it has stopped successfully.